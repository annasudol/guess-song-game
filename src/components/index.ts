export * from './alert/Alert';
export * from './button/Button';
export * from './button/ButtonLogout';
export * from './image/Image';
export * from './game/Game';
export * from './login/Login';
export * from './listSongs/ListSongs';
export * from './navigation/Navigation';
export * from './errorMessage/ErrorMessage';
export * from './playlists/Playlists';
export * from './playListItem/PlaylistItem';
export * from './scoreInfo/ScoreInfo';
export * from './scores/Scores';
export * from './summary/Summary';
export * from './search/Search';
export * from './song/Song';
export * from './timer/Timer';
