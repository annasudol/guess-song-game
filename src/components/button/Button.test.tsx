import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { Button } from '../../components';

describe('component', (): void => {
  describe('button', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(<Button>test</Button>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('match snapshot when href', (): void => {
      const wrapper = shallow(
        <Button href={{ path: '/test', target: true }} spacing={{ mt: '10' }}>
          test
        </Button>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('match snapshot with mainBtn class', (): void => {
      const wrapper = shallow(<Button mainBtn>test</Button>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
