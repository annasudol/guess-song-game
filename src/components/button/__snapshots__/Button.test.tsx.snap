// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component button match snapshot 1`] = `
<button
  className="btn font-alba"
>
  test
</button>
`;

exports[`component button match snapshot when href 1`] = `
<a
  className="btn mt-10 font-alba"
  href="/test"
  target="_blank"
>
  test
</a>
`;

exports[`component button match snapshot with mainBtn class 1`] = `
<button
  className="btn--game bg-transparent focus:outline-none hover:bg-orange-500 w-full md:w-5/12 text-orange-500 font-semibold hover:text-indigo-900 py-2 border border-orange-500 hover:border-transparent rounded m-1 mt-4"
>
  test
</button>
`;
